lib/mkFakeDerivation.nix

```diff
@@ -9,10 +12,47 @@
       meta = {outputsToInstall = ["out"];} // (publishData.eval.meta or {});
       outputs = publishData.eval.outputs or {"out" = lib.head publishData.element.storePaths;};
     };
-  fromSource = with publishData.element;
-    if url == ""
-    then throw "url = \"\" so this fakeDerivation can't be built from source. Note that fakeDerivations created from self cannot be built from source"
-    else lib.getAttrFromPath attrPath (builtins.getFlake url);
+  fromSource = let
+    fromNixpkgs = builtins.match "github:flox/nixpkgs/.*" publishData.element.url != null;
+    # implement allowedUnfreePackages as allowUnfreePredicate, but still honor
+    # allowUnfreePredicate
+    allowUnfreePredicate =
+      if context ? config.nixpkgs-config.allowUnfreePredicate
+      then
+        if context ? config.nixpkgs-config.allowedUnfreePackages
+        then throw "only one of allowedUnfreePackages and allowUnfreePredicate can be specified in config.nixpkgs-config"
+        else context.config.nixpkgs-config.allowUnfreePredicate
+      else if context ? config.nixpkgs-config.allowedUnfreePackages
+      then pkg: builtins.elem (lib.getName pkg) context.config.nixpkgs-config.allowedUnfreePackages
+      else _: false;
+    # based on stdenv check-meta.nix
+    hasDeniedUnfreeLicense =
+      eval.meta.unfree
+      && !(context.config.nixpkgs-config.allowUnfree or false)
+      && !(allowUnfreePredicate eval.pname);
+  in
+    if
+      fromNixpkgs
+      && hasDeniedUnfreeLicense
+    then throw "Encountered unfree package ${eval.pname} that is not allowed; add exceptions to the flake's config.nixpkgs-config"
+    else if fromNixpkgs
+    then let
+      pkgSystem =
+        if builtins.head publishData.element.attrPath != "legacyPackages"
+        then throw "first element of attrPath in nixpkgs not equal to legacyPackages"
+        else builtins.elemAt publishData.element.attrPath 1;
+      nixpkgsFlake = builtins.getFlake publishData.element.url;
+      pkgs = import nixpkgsFlake {
+        config = (builtins.removeAttrs context.config.nixpkgs-config ["allowedUnfreePackages"]) // {inherit allowUnfreePredicate;};
+        system = pkgSystem;
+      };
+    in
```

Comment:
why don't we reuse check-meta directly?

assuming context is system specialized

``` 
          maybe 
          vvvvvvvvvvvvvvvvvvvvvvv
import "${context.nixpkgs.outPath}/pkgs/stdenv/generic/check-meta.nix" {
  inherit (context.nixpkgs) lib config;
  # Nix itself uses the `system` field of a derivation to decide where
  # to build it. This is a bit confusing for cross compilation.
  inherit (context.nixpkgs.stdenv) hostPlatform;
};
```

could work and give us all the stdenv checking at once.

Reply:
Yeah that's just conventional upstream. No reason we can't do something better

Reply:
We already have all the stdenv checking. It's just a matter of:
```
doFloxChecking
checkMeta
```
vs
```
try CheckMeta
catch
doFloxStuff
```

Is there any advantage to the latter?